### PR TITLE
Fix LAN discovery when UDP is disabled

### PR DESCRIPTION
**Pull request by anthonybilinski**
Before, LAN discovery could not send broadcast packets when UDP was disabled, because INET family was not set.

---
This change is [Reviewable](https://reviewable.io/reviews/toktok/c-toxcore/866)

### CODE REVIEW

**Comment by anthonybilinski** [comment]
I guess this is still sending UDP packets when UDP is disabled.

I feel like when someone "disables UDP" it's either to opt out of DHT for battery/traffic reasons (mobile), or to opt out of DHT for proxy reasons, and not really because of the protocol of UDP vs TCP. So it still seems like allowing LAN discovery when UDP is disabled is desirable, even if a little confusing.

**Comment by iphydf** [comment]
:lgtm_strong:

---

Reviewed 3 of 3 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://beta.reviewable.io/reviews/toktok/c-toxcore/866#-:-LA9HxR94lbt-LvDn7wJ:b-jbsf8z)*

**Comment by fcore117** [comment]
anthonybilinski: Could this be done using TCP discovery without using UDP when UDP is disabled?

This feature could be really useful when not wanting to relay because of high data use and connection count(UDP on) but can communicate directly in LAN network still. Helps avoid killing cheap routers too when lot of Tox users are behind same router.

**Comment by anthonybilinski** [comment]
On review I don't think this is a good PR, please don't merge.

I need to look into behaviour and test more, but I think with my change, it's still bootstrapping off nodes got from peer discovery on LAN, which is effectively ignoring UDP disabled setting. Meaning if you have UPD disabled and LAN enabled, you'll quickly be sending UDP out to a bunch of internet nodes that you discovered from nodes on your LAN, and acting equivalently to having UDP disabled.

Basically the feature I wanted to add was to control between 1) Full global DHT mode, 2) TCP relay mode, don't participate in DHT, and independently 3) look for peers on local network.

With 2) and 3) together, I think local net peers should be connected to (even using UDP), but peers found through local discovery off local net should not be connected to. TCP relays found through local peer discovery should still be connected to.

**Comment by fcore117** [comment]
iphydf: TCP discovery > LAN discovery using TCP only.

**Comment by iphydf** [comment]
@fcore117 how does that work?

**Comment by iphydf** [comment]
@anthonybilinski please reopen, or open a new PR when you have something you'd like to have merged.
